Homework №3 from 21.09.2022/task3.c

```diff
@@ -0,0 +1,130 @@
+#include <stdio.h>
+#include <stdlib.h>
+#define arrTestLen 9
+
+void printArr(int *array, int lenOfArray)
+{
+    for (int i = 0; i < lenOfArray; ++i)
+    {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+void swap(int *firstElement, int *secondElement)
+{
+    int temp = *secondElement;
+    *secondElement = *firstElement;
+    *firstElement = temp;
+}
+
+int divide(int *array, int left, int right)
+{
+    int changingElem = left;
+    for (int i = left; i < right; ++i)
+    {
+        if (array[i] < array[right])
+        {
+            swap(&array[changingElem], &array[i]);
+            changingElem += 1;
+        }
+    }
+    swap(&array[right], &array[changingElem]);
+    return changingElem;
+}
+
+void quickSortRec(int *array, int left, int right)
+{
+    if (left < right)
+    {
+        int centerElem = divide(array, left, right);
+        quickSortRec(array, left, centerElem - 1);
+        quickSortRec(array, centerElem + 1, right);
+    }
+}
+
+void quickSort(int *array, int lenOfArray)
+{
+    quickSortRec(array, 0, lenOfArray - 1);
+}
+
+void mostPopularElement(int *array, int lenOfArray, int *subArr)
```

Comment:
```suggestion
void mostPopularElement(int *array, int lenOfArray, int *elementPointer, int *countPointer)
```
Заворачивать несколько ответов в массив - так себе идея. Либо два указателя, кудадва результата положить можно. Либо завести структуру с двкмя полями. и структуру возвращать.
Но вообще в задании не требовалось вернуть частоту элемента.